util/FileCreator.cpp

```diff
@@ -197,7 +198,16 @@ int FileCreator::openExistingFile(ThreadCtx &threadCtx,
   return res;
 }
 
-int FileCreator::createFile(ThreadCtx &threadCtx, const string &relPathStr) {
+int openWithPerm(const char *path, int oFlags, int perm) {
+  if (perm >= 512) {
```

Comment:
I am not sure we need this check against 512. Just set the default value of BlockDetails::permission to 0644.